storage/src/test/java/tech/pegasys/teku/storage/server/AbstractDatabaseTest.java

```diff
@@ -712,8 +718,10 @@ protected void assertHotBlocksAndStates(
 
       final List<BeaconState> hotStates =
           currentStore.getBlockRoots().stream()
-              .map(currentStore::getBlockState)
-              .filter(Objects::nonNull)
+              .map(currentStore::retrieveBlockState)
+              .filter(SafeFuture::isDone)
```

Comment:
Not sure I'm getting what is this filter for? Why not just `join`?

Reply:
I was thinking that we could avoid blocking in this way, and if some futures were skipped because they haven't resolved the assertions below should fail.  But on second thought, I think that's a bad assumption - will rework.

Reply:
May be just set the test timeout?

Reply:
Ended up just adding an assert inside of the `map` function